### PR TITLE
fix(build): lock madsim version

### LINKED ISSUE #2782: ci: docker build failed with `detach` not found
```
#19 174.6 error[E0599]: no method named `detach` found for struct `tokio::task::JoinHandle` in the current scope
1957
#19 174.6    --> src/stream/src/task/stream_manager.rs:569:26
1958
#19 174.6     |
1959
#19 174.6 569 |                         .detach();
1960
#19 174.6     |                          ^^^^^^ method not found in `tokio::task::JoinHandle<()>`
1961
#19 174.6 
1962
#19 174.9 For more information about this error, try `rustc --explain E0599`.
1963
#19 174.9 error: could not compile `risingwave_stream` due to previous error
1964
#19 174.9 warning: build failed, waiting for other jobs to finish...
```

I guess this is somehow related to madsim? cc @wangrunji0408

### PR DESCRIPTION
**Pull request by wangrunji0408**
Signed-off-by: Runji Wang <wangrunji0408@163.com>

## What's changed and what's your intention?

Lock madsim version to `0.2.0-alpha.1` to fix the build.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
fix #2782

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
LGTM! So `Cargo.lock` is ignored from Dockerfile? I'll dig into that later.
